javascript/ql/src/semmle/javascript/security/dataflow/TaintedPathCustomizations.qll

```diff
@@ -355,6 +355,33 @@ module TaintedPath {
     }
   }
 
+  /**
+   * An expression of form `isInside(x, y)` or similar, where `isInside` is
+   * a library check for the relation between `x` and `y`.
+   */
+  class IsInsideCheckSanitizer extends DataFlow::BarrierGuardNode {
+    DataFlow::Node checked;
+
+    IsInsideCheckSanitizer() {
+      exists(string name, DataFlow::CallNode check |
+        name = "path-is-inside" or
+        name = "is-path-inside"
+      |
+        check = DataFlow::moduleImport(name).getACall() and
+        checked = check.getArgument(0) and
+        check = this
+      )
+    }
+
+    override predicate blocks(boolean outcome, Expr e) {
+      e = checked.asExpr() and
+      outcome = true
+      // Note that `is-path-inside` states:
+      // > "Note that relative paths are resolved against process.cwd() to make them absolute."
+      // so it may be possible to undo this sanitization by later prepending some other path to the sanitized path.
```

Comment:
Did you mean to leave this comment here? Doesn't seem like something we need to worry about.